### DISCUSSION TITLE
How do I catch paste event from the right click menu?

### DISCUSSION THREAD

**Discussion by electroglyph** [sourcelair/xterm.js#5072] [Q&A]
no stdin...

**Comment by electroglyph** [comment]
To help any future searchers:

To detect a right-click paste the method I used is: from onData event look at data length.  if the length is > 1 and the first char isn't 0x1B (i.e. it isn't a control code) then assume a paste is happening.  it's a bit hacky and i bet this doesn't cover all cases, but it's mostly effective for me

**Reply by jerch** [reply]
Plz note that your heuristic most likely will fail for very fast typed keyboard input - there is a small chance, that fast keyboard succession will be been in one `onData` event.
